atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -386,7 +386,7 @@ private void lowerConsistencyWhenSafe() {
                                         colFam,
                                         pred,
                                         readConsistency);
-                                Map<Cell, Value> ret = Maps.newHashMap();
+                                Map<Cell, Value> ret = Maps.newHashMapWithExpectedSize(batch.size());
```

Comment:
I think your change is safe, and I'd wager 95%+ is likely to have ret.size >= batch.size. The only cases where I think this could bite us are if one were to do negative lookups, i.e. Checking non-existence where you pass in a bunch of keys and happy path is 0 results.